### PR TITLE
Separate frontend and backend in development setup

### LINKED ISSUE #2266: Serve Frontend Code with Node/Nginx
We should consider treating the backend server as a RESTful API only. Hence, all frontend code should be served by a dedicated tool for handling JS/CSS/HTML assets, e.g. NGinx. 
For frontend dev we could contemplate to use a hot-reloading NodeJS server for development.

ToDos:
- Check which routes are still served by Play Framework / Scala
- Move configuration options from `pageSkelton.html.scala` to new hosting solution
- Add hot-reloading dev server for React
- Restructure the folder hierarchy accordingly. e.g. move frontend code out of `app/assets`. Update paths in build tools.

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/2266/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by youri-k**
- in the development setup sbt no longer handles the webpack compilation
- the normal yarn command remains `yarn start` which starts a proxy, that in turn starts a `webpack-dev-server` and a `backend-server`
- for backend development you can run `yarn start-backend` which builds the frontend code once and then starts `webknossos`

### URL of deployed dev instance (used for testing):
- https://separatefrontendbackend.webknossos.xyz

### Steps to test:
- test everything

### Issues:
- fixes #2266 

------
~~- [ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
~~- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
~~- [ ] Updated [documentation](../blob/master/docs) if applicable~~
~~- [ ] Needs datastore update after deployment~~
- [ ] Ready for review

### CODE REVIEW

File: app/assets/javascripts/dashboard/advanced_dataset/dataset_action_view.js [hunk 1]
```diff
@@ -56,12 +56,8 @@ class DatasetActionView extends React.PureComponent<Props, State> {
 
     const volumeTracingMenu = (
       <Dropdown overlay={menu} trigger={["click"]}>
-        <a href="#" title="Create Volume Tracing">
-          <img
-            src="/assets/images/volume.svg"
-            alt="volume icon"
-            style={centerBackgroundImageStyle}
-          />{" "}
+        <a href="#" title="Create Volume tracing">
```
Comment (daniel-wer) [inline app/assets/javascripts/dashboard/advanced_dataset/dataset_action_view.js]: You made this a small `t` by accident.
```suggestion
        <a href="#" title="Create Volume Tracing">
```
You'll need to recreate the snapshots again then, unfortunately ^.~

**Comment by philippotto** [comment]
Awesome! Thanks for pulling through with this task :)

> for backend development you can run yarn start-backend which builds the frontend code once and then starts webknossos

Is there a corresponding command for frontend development? As far as I understand, `yarn start` itself runs sbt as usual with incremental compiling enabled. It's probably not really necessary, since there should be no re-compilation anyway if nothing changed, but that's true for frontend code when doing backend development, too. So, why is this asymmetrical?

File: project/AssetCompilation.scala [hunk 2]
```diff
@@ -20,120 +19,103 @@ object AssetCompilation {
...
 
   private def killProcess(pid: String) = {
-    if(isWindowsSystem)
+    if (isWindowsSystem)
       Process("kill" :: "-f" :: pid :: Nil).run()
     else
       Process("kill" :: pid :: Nil).run()
   }
 
   private def npmInstall: Def.Initialize[Task[Seq[File]]] = Def task {
```
[the remark this answers is not in the document]
Comment (fm3) [reply project/AssetCompilation.scala]: (somewhat unfortunately) it still needs npm install to run the diff schema node script

**Comment by youri-k** [comment]
> Is there a corresponding command for frontend development? As far as I understand, `yarn start` itself runs sbt as usual with incremental compiling enabled. It's probably not really necessary, since there should be no re-compilation anyway if nothing changed, but that's true for frontend code when doing backend development, too. So, why is this asymmetrical?
@philippotto AFAIK you can't switch off the incremental compiler of sbt [source](https://stackoverflow.com/questions/23226114/can-the-incremental-compiler-in-sbt-be-turned-off-for-ci-builds) and it doesn't seem like a huge problem. If you think this needs to be addressed further I can look into it.
I introduced the backend command because there is no need to run a webpack-dev-server and minimize the overhead.

**Comment by youri-k** [comment]
@fm3 @jstriebel @rschwanhold I would be glad if you could review this from a backend perspective but I understand that there might be more important tasks :slightly_smiling_face:

**Pull request merged**